tests/Microsoft.Bot.Builder.Core.Extensions.Tests/StorageBaseTests.cs

```diff
@@ -199,6 +199,23 @@ public class TypedObject
         {
             await storage.Delete("unknown_key");
         }
+
+        protected async Task _batchCreateObjectTest(IStorage storage)
+        {
+            var storeItemsList = new List<Dictionary<string, object>>(new[]
+                {
+                new Dictionary<string, object> {["createPoco"] = new PocoItem() { Id = "1", Count = 0 }},
+                new Dictionary<string, object> {["createPoco"] = new PocoItem() { Id = "1", Count = 1 }},
+                new Dictionary<string, object> {["createPoco"] = new PocoItem() { Id = "1", Count = 2 }}
+            });
+
+            await Task.WhenAll(storeItemsList.Select(storeItems => storage.Write(storeItems)));
```

Comment:
For the record, this does not fully test parallel/concurrent access. You should throw a `Task.Run` around each one to force the entire implementation of `Write` to execute on a separate thread.

Reply:
Agree, we started with the first version that validated the erroneous behavior. We'll change the test before moving forward.